src/docstub/doctype.lark

```diff
@@ -1,11 +1,12 @@
 ?start : doctype
 
-doctype : type_or ("," optional)? ("," extra_info)?
+doctype : (literals | type_or) ("," optional)? ("," extra_info)?
```

Comment:
See https://github.com/scientific-python/docstub/pull/2/commits/5a288281acd715ce7c1d92acb4ba935fd928d57e.

Reply:
I had never seen nor considered that option, but now thinking about it there are a couple places I could use it. If you use it or feel strongly about it maybe we could use something similar to arrays for mappings in the sense a subset of names are allowed, and only if they are present can then curly brackets indicate two subtypes with colon. My guess is dict and mapping alone will cover 90% of the cases, maybe mutablemapping could also be there.

Plus a way to extend those names for both dict and array (to allow tensor for example in projects that use it)

Reply:
> maybe we could use something similar to arrays for mappings in the sense a subset of names are allowed

I think it might be more confusing if we restricted who can use the `mapping of {KT: VT}` syntax? :thinking:

Reply:
Happy to keep literals as top level option only